API.md

```diff
@@ -0,0 +1,106 @@
+## Models
+
+### Blueprint
+TBD
+
+### Header
+
+```
+{
+	type:      'header';
+	id:        number;
+	sourceUrl: string; // URL of the original page from which the header was created.
+	// More fields TBD.
+}
+```
+### Footer
+
+```
+{
+	type:      'footer';
+	id:        number;  
+	sourceUrl: string; // URL of the original page from which the footer was created.
+	// More fields TBD.
+}
+```
+### BlogPost
+
+```
+{
+	type:          'blog-post';
+	id:            number;  
+	transformedId: number;      // Id of the corresponding transformed post
+	sourceUrl:     string;      // URL of the source page from which this blog post was created.
+	date: {
+		type:     'date';
+	    original: string; // original HTML
+	    parsed:   string; // ISO date string
+	},
+	title: {
+		type:     'text';
+	    original: string; // original HTML
+	    parsed:   string; // plain text
+	},
+	content: {
+		type:     'html';
+	    original: string; // original HTML
+	    parsed:   string; // block markup
+	}
```

Comment:
I think it's best to keep 2 representation objects to keep things simple. This one being the representation frontend works with and a flatten object representation for api interaction. We would continue to have a function that converts objects in both directions.